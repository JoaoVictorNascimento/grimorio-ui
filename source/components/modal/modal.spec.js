import '../../../internals/test/helper';
jest.mock('../svg')
import Modal from './modal-component';
import Button from '../button';

/** @test {Modal} */
describe('Modal component', () => {

  const emptyProps = {
    children: undefined,
    open: undefined,
    showClose: undefined,
    closeOnOverlay: undefined,
    type: undefined,
    confirmInverted: undefined,
    showButton: undefined,
    innerPadding: undefined,
    message: undefined,
    confirmButtonText: undefined,
    cancelButtonText: undefined,
    onClose: undefined,
    onConfirm: undefined,
  }

  let wrapper
  beforeAll(() => {
    wrapper = shallow(
      <Modal {...emptyProps} />
    );
  })

  describe('#render', () => {
    it('should render modal with children', () => {
      wrapper.setProps({ ...emptyProps, children: <div>teste com filho</div> })
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe('render success modal', () => {

      beforeAll(() => {
        wrapper.setProps({
          ...emptyProps,
          type: 'success',
          message: 'mensagem do modal de sucesso',
        })
      })
      
      it('should render right', () => {
        expect(wrapper.html()).toMatchSnapshot();
      })

      it('when receives showButton props, button should call function onClose onClick', () => {
        const props = {
          onClose: jest.fn()
        }
        const spy = jest.spyOn(props, 'onClose')
        wrapper.setProps({showButton: true, onClose: props.onClose})
        wrapper.find('.actions').find(Button).props().onClick()
      expect(spy).toHaveBeenCalled()
      })
      
    });

    describe('render fail modal', () => {

      beforeAll(() => {
        wrapper.setProps({
          ...emptyProps,
          type: 'fail',
          message: 'Ops, algo deu errado :(',
        })
      })
      
      it('should render right', () => {
        expect(wrapper.html()).toMatchSnapshot();
      })

      it('when receives showButton props, button should call function onClose onClick', () => {
        const props = {
          onClose: jest.fn()
        }
        const spy = jest.spyOn(props, 'onClose')
        wrapper.setProps({showButton: true, onClose: props.onClose})
        wrapper.find('.actions').find(Button).props().onClick()
      expect(spy).toHaveBeenCalled()
      })
    });

    describe('should render confirm modal', () => {
     
      beforeAll(() => {
        wrapper.setProps({
          ...emptyProps,
          type: 'confirm',
          message: 'Mensagem do modal de confirm',
          confirmButtonText: 'Tenho certeza'
        })
      })
      
      it('should render right', () => {
        expect(wrapper.html()).toMatchSnapshot();
      })

      it('should call function onClose onClick button close', () => {
        const props = {
          onClose: jest.fn()
        }
        const spy = jest.spyOn(props, 'onClose')
        wrapper.setProps({onClose: props.onClose})
        wrapper.find('.actions').find(Button).at(0).props().onClick()
        expect(spy).toHaveBeenCalled()
      })

      it('should call function onConfirm onClick button close', () => {
        const props = {
          onConfirm: jest.fn()
        }
        const spy = jest.spyOn(props, 'onConfirm')
        wrapper.setProps({onConfirm: props.onConfirm})
        wrapper.find('.actions').find(Button).at(1).props().onClick()
        expect(spy).toHaveBeenCalled()
      })
    });

    describe('when receives showClose props', () => {
      let spy
      beforeAll(() => {
        const props = {
          onClose: jest.fn()
        }
        spy = jest.spyOn(props, 'onClose')
        wrapper.setProps({
          ...emptyProps,
          showClose: true,
          onClose: props.onClose
        })
      })
      
      it('should render icon to close', () => {
        expect(wrapper.find('.close')).toHaveLength(1);
      })
      it('should call onClose props when clicks on icon close', () => {
        wrapper.find('.close').props().onClick()
        expect(spy).toHaveBeenCalled();
      })
    })

    it('When receives closeOnOverlay props, should call onClose function', () => {
      const props = {
        onClose: jest.fn()
      }
      const spy = jest.spyOn(props, 'onClose')
      wrapper.setProps({
        ...emptyProps,
        closeOnOverlay: true,
        onClose: props.onClose
      })
      wrapper.find('.overlay').simulate('click')
      expect(spy).toHaveBeenCalled();
    })
  });
});
