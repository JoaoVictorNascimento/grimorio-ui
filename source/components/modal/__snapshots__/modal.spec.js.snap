// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component #render render fail modal should render right 1`] = `
"<div className=\\"wrap\\">
  <div className=\\"modal\\" type=\\"fail\\" confirmInverted={false} showButton={false} message=\\"Ops, algo deu errado :(\\" confirmButtonText=\\"Okay\\" cancelButtonText=\\"Cancelar\\" onConfirm={[undefined]}>
    <Icon onClick={[undefined]} className=\\"close\\" size=\\"20\\" name=\\"close\\" align=\\"middle\\" />
    <div className=\\"content\\">
      <div className=\\"base\\">
        <div className=\\"imageWrap\\">
          <Svg src=\\"fail\\" width=\\"50px\\" height=\\"50px\\" align=\\"middle\\" />
        </div>
        <div className=\\"message\\">
          Ops, algo deu errado :(
        </div>
      </div>
    </div>
  </div>
  <div onClick={[undefined]} className=\\"overlay\\" />
</div>"
`;

exports[`Modal component #render render success modal should render right 1`] = `
"<div className=\\"wrap\\">
  <div className=\\"modal\\" type=\\"success\\" confirmInverted={false} showButton={false} message=\\"mensagem do modal de sucesso\\" confirmButtonText=\\"Okay\\" cancelButtonText=\\"Cancelar\\" onConfirm={[undefined]}>
    <Icon onClick={[undefined]} className=\\"close\\" size=\\"20\\" name=\\"close\\" align=\\"middle\\" />
    <div className=\\"content\\">
      <div className=\\"base\\">
        <div className=\\"imageWrap\\">
          <Svg src=\\"success\\" width=\\"70px\\" height=\\"70px\\" align=\\"middle\\" />
        </div>
        <div className=\\"message\\">
          mensagem do modal de sucesso
        </div>
      </div>
    </div>
  </div>
  <div onClick={[undefined]} className=\\"overlay\\" />
</div>"
`;

exports[`Modal component #render should render confirm modal should render right 1`] = `
"<div className=\\"wrap\\">
  <div className=\\"modal\\" type=\\"confirm\\" confirmInverted={false} showButton={false} message=\\"Mensagem do modal de confirm\\" confirmButtonText=\\"Tenho certeza\\" cancelButtonText=\\"Cancelar\\" onConfirm={[undefined]}>
    <Icon onClick={[undefined]} className=\\"close\\" size=\\"20\\" name=\\"close\\" align=\\"middle\\" />
    <div className=\\"content\\">
      <div className=\\"base\\">
        <div className=\\"imageWrap\\">
          <Svg src=\\"fail\\" width=\\"50px\\" height=\\"50px\\" align=\\"middle\\" />
        </div>
        <div className=\\"message\\">
          Mensagem do modal de confirm
        </div>
        <div className=\\"actions\\">
          <Button className=\\"button\\" color=\\"variant\\" modifier=\\"outline\\" onClick={[undefined]} disabled={false} size=\\"medium\\" type=\\"button\\" active={false}>
            Cancelar
          </Button>
          <Button className=\\"button\\" color=\\"primary\\" modifier={{...}} onClick={[undefined]} disabled={false} size=\\"medium\\" type=\\"button\\" active={false}>
            Tenho certeza
          </Button>
        </div>
      </div>
    </div>
  </div>
  <div onClick={[undefined]} className=\\"overlay\\" />
</div>"
`;

exports[`Modal component #render should render modal with children 1`] = `
"<div className=\\"wrap\\">
  <div className=\\"modal\\" type=\\"custom\\" confirmInverted={false} showButton={false} message={[undefined]} confirmButtonText=\\"Okay\\" cancelButtonText=\\"Cancelar\\" onConfirm={[undefined]}>
    <Icon onClick={[undefined]} className=\\"close\\" size=\\"20\\" name=\\"close\\" align=\\"middle\\" />
    <div className=\\"content\\">
      <div>
        teste com filho
      </div>
    </div>
  </div>
  <div onClick={[undefined]} className=\\"overlay\\" />
</div>"
`;
